### PR TITLE
Add Math::TAU

### LINKED ISSUE #10165: Add constant Math::TAU
This is a simple proposal to add the math constant Math::TAU = 2*Math::PI.

See: https://tauday.com/

TAU has been included in a growing number of languages (D, C#, F#, Nim, Zig, Rust, Python3, Julia, et al), and this would add Crystal to that list, and make it even cooler. :-)

### PR DESCRIPTION
**Pull request by j8r**
Resolves #10165

### CODE REVIEW

File: src/math/math.cr [hunk 1]
```diff
@@ -3,7 +3,8 @@ require "./libm"
 module Math
   extend self
 
-  PI    = 3.14159265358979323846
+  PI    =  3.14159265358979323846
+  TAU   = 6.283185307179586476925
```
Comment (Sija) [inline src/math/math.cr:7]:
```suggestion
  TAU   = PI * 2
```

Comment (asterite) [reply src/math/math.cr:7]: Please no. I'm not sure the compiler will be able to inline it like that.

Comment (j8r) [reply src/math/math.cr:7]: I was expecting this, and that's what I thought initially. That mathematically correct, but with how computers work, it does not produce a correct result.
```cr
p Math::PI * 2 #=> 6.283185307179586
```
`6.283185307179586476925` != `6.283185307179586`.
Similarly, `Math::TAU / 2` != `Math::PI`.

I have checked how this constant in implemented in C#, it is done like this.

Comment (j8r) [reply src/math/math.cr:7]: Technically, the result may be correct, because it is a rounded number, but at the end is we lose precision.

Comment (j8r) [reply src/math/math.cr:7]: Technically, the result may be correct, because it is a rounded number, but the point is at the end, precision is lost.

Comment (straight-shoota) [reply src/math/math.cr:7]: In any case, the obviously only correct solution would be `PI = TAU / 2` 😄

**Comment by straight-shoota** [comment]
Would you mind adding a short doc comment? Something like https://doc.rust-lang.org/nightly/std/f64/consts/constant.TAU.html

**Review by straight-shoota — APPROVED** [review]

File: src/math/math.cr [hunk 2]
```diff
@@ -2,8 +2,11 @@ require "./libm"
 
 module Math
   extend self
-
-  PI    = 3.14159265358979323846
+  # Archimedes' constant (π).
```
Comment (Sija) [inline src/math/math.cr:5]:
```suggestion

  # Archimedes' constant (π).
```

Comment (j8r) [reply src/math/math.cr:6]: Thanks, applied.

**Review by jhass — APPROVED** [review]

**Pull request merged**